src/gui/applicationpalette.cpp

```diff
@@ -0,0 +1,333 @@
+ * delete this exception statement from your version.
+ */
+ 
+#include "applicationpalette.h"
+
+#include <QApplication>
+#include <QPalette>
+
+QColor ApplicationPalette::window()
+{
+    return qApp->palette().color(QPalette::Window);
+}
+
+QColor ApplicationPalette::windowText()
+{
+    return qApp->palette().color(QPalette::WindowText);
+}
+
+QColor ApplicationPalette::windowTextDisabled()
+{
+    return qApp->palette().color(QPalette::Disabled, QPalette::WindowText);
+}
+
+QColor ApplicationPalette::base()
+{
+    return qApp->palette().color(QPalette::Base);
+}
+
+QColor ApplicationPalette::alternateBase()
+{
+    return qApp->palette().color(QPalette::AlternateBase);
+}
+
+QColor ApplicationPalette::text()
+{
+    return qApp->palette().color(QPalette::Text);
+}
+
+QColor ApplicationPalette::textDisabled()
+{
+    return qApp->palette().color(QPalette::Disabled, QPalette::Text);
+}
+
+QColor ApplicationPalette::toolTipBase()
+{
+    return qApp->palette().color(QPalette::ToolTipBase);
+}
+
+QColor ApplicationPalette::toolTipText()
+{
+    return qApp->palette().color(QPalette::ToolTipText);
+}
+
+QColor ApplicationPalette::toolTipTextDisabled()
+{
+    return qApp->palette().color(QPalette::Disabled, QPalette::ToolTipText);
+}
+
+QColor ApplicationPalette::brightText()
+{
+    return qApp->palette().color(QPalette::BrightText);
+}
+
+QColor ApplicationPalette::brightTextDisabled()
+{
+    return qApp->palette().color(QPalette::Disabled, QPalette::BrightText);
+}
+
+QColor ApplicationPalette::highlight()
+{
+    return qApp->palette().color(QPalette::Highlight);
+}
+
+QColor ApplicationPalette::highlightedText()
+{
+    return qApp->palette().color(QPalette::HighlightedText);
+}
+
+QColor ApplicationPalette::highlightedTextDisabled()
+{
+    return qApp->palette().color(QPalette::Disabled, QPalette::HighlightedText);
+}
+
+QColor ApplicationPalette::button()
+{
+    return qApp->palette().color(QPalette::Button);
+}
+
+QColor ApplicationPalette::buttonText()
+{
+    return qApp->palette().color(QPalette::ButtonText);
+}
+
+QColor ApplicationPalette::buttonTextDisabled()
+{
+    return qApp->palette().color(QPalette::Disabled, QPalette::ButtonText);
+}
+
+QColor ApplicationPalette::link()
+{
+    return qApp->palette().color(QPalette::Link);
+}
+
+QColor ApplicationPalette::linkVisited()
+{
+    return qApp->palette().color(QPalette::LinkVisited);
+}
+
+QColor ApplicationPalette::light()
+{
+    return qApp->palette().color(QPalette::Light);
+}
+
+QColor ApplicationPalette::midlight()
+{
+    return qApp->palette().color(QPalette::Midlight);
+}
+
+QColor ApplicationPalette::mid()
+{
+    return qApp->palette().color(QPalette::Mid);
+}
+
+QColor ApplicationPalette::dark()
+{
+    return qApp->palette().color(QPalette::Dark);
+}
+
+QColor ApplicationPalette::shadow()
+{
+    return qApp->palette().color(QPalette::Shadow);
+}
+
+void ApplicationPalette::setWindow(const QColor &c)
+{
+    QPalette p = qApp->palette();
```

Comment:
I would extract this common pattern (these 3-lines) to a helper function (put it in anonymous namespace) and call it instead.

Reply:
done.